webapp-client/src/component/AddCategoryForm.js

```diff
@@ -1,32 +1,73 @@
 import React from 'react';
+import ErrorMessage from './ErrorMessage';
 
 export default class AddCategoryForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hasError: false,
+    };
+    this.createCategory = this.createCategory.bind(this);
+    this.clearHasError = this.clearHasError.bind(this);
```

Comment:
Optional: consider adding a comment in all components explaining why you're binding here? Maybe you can link to https://reactjs.org/docs/handling-events.html

If you do decide to do this, then for this PR prob just do it for the components already in this PR.